### ISSUE TITLE
How does WAN num_repeats effect the total step count?

### ISSUE THREAD

**Issue by Basedbase-ai** [tdrussell/diffusion-pipe#188]
I have num_repeats set to 3 with a dataset of 104 videos and im up to step 70 and im not even done with epoch 1. Does repeats set to 3 mean its 312 steps per epoch? If so are repeats even necessary for a large dataset to produce a good lora?

**Comment by tdrussell** [comment]
>Does repeats set to 3 mean its 312 steps per epoch

Yes, as long as batch size and gradient accumulation steps are 1 (i.e. as single step is just 1 video).

Repeats are generally only useful with smallish datasets and aspect ratio bucket training, so that less examples get dropped. For each size bucket, it must round down the examples to the nearest multiple of the global batch size. If you have a very small dataset, this could mean a significant fraction gets dropped, but num_repeats makes it act like the dataset is larger (it is as if you are copy-pasting the examples and duplicating them that many times).
